benchmark/buckets_benchmark.cpp

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2020 Scipp contributors (https://github.com/scipp)
+/// @file
+#include <benchmark/benchmark.h>
+
+#include "scipp/dataset/bucket.h"
+#include "scipp/dataset/dataset.h"
+#include "scipp/dataset/shape.h"
+#include "scipp/variable/bucket_model.h"
+#include "scipp/variable/operations.h"
+
+using namespace scipp;
+
+auto make_buckets(const scipp::index size, const scipp::index count) {
+  using Model = variable::DataModel<bucket<DataArray>>;
+  Dimensions dims{Dim::Y, size};
+  Variable indices = makeVariable<std::pair<scipp::index, scipp::index>>(dims);
+  scipp::index current = 0;
+  for (auto &range : indices.values<std::pair<scipp::index, scipp::index>>()) {
+    range.first = current;
+    current += count / size;
+    range.second = current;
+  }
+  Variable data = makeVariable<double>(Dims{Dim::X}, Shape{count});
+  DataArray buffer = DataArray(data, {{Dim::X, data + data}});
+  return Variable{
+      std::make_unique<Model>(std::move(indices), Dim::X, std::move(buffer))};
+}
+
+static void BM_buckets_concatenate(benchmark::State &state) {
+  const scipp::index nBucket = state.range(0);
+  const scipp::index nEvent = state.range(1);
+  auto events = make_buckets(nBucket, nEvent);
+  for (auto _ : state) {
+    auto var = dataset::buckets::concatenate(events, events);
+    state.PauseTiming();
```

Comment:
Wouldn't you save more lines of code if you used `benchmark::DoNotOptimize`?

Reply:
I didn't to this to avoid optimization, rather to avoid inclusion of deallocation in the timings.